see https://gerrit.libreoffice.org/#/c/49614/.

sysui/desktop/apparmor/program.senddoc
Line 19:
why do we need to allow dbus, chrome, .. here?

See https://sources.debian.org/src/apparmor/2.12-2/profiles/apparmor.d/abstractions/ubuntu-helpers/.

(Besides that I don't like the "ubuntu" there at all, but that is another story)

sysui/desktop/apparmor/program.senddoc
Line 19:
The other (easy) option is to have xdg-* just go to unconfined.  I'm not sure there will be a huge difference in security.

The initial version got merged to libreoffice-6-0 without the master one being
merged...

diff --git a/sysui/desktop/apparmor/program.senddoc b/sysui/desktop/apparmor/program.senddoc
index 4ee34d01c941..4392c757ed26 100644
--- a/sysui/desktop/apparmor/program.senddoc
+++ b/sysui/desktop/apparmor/program.senddoc
@@ -14,7 +14,6 @@
 
 profile libreoffice-senddoc INSTDIR-program/senddoc {
   #include <abstractions/base>
-  #include <abstractions/ubuntu-helpers>
 
   owner /tmp/lu**       rw,    #makes files like luRRRRR.tmp/lubRRRR.tmp where R is random
                                #Note, usually it's lub or luc, don't know why.
@@ -26,8 +25,8 @@ profile libreoffice-senddoc INSTDIR-program/senddoc {
   /usr/bin/basename     rmix,
   /{usr/,}bin/grep      rmix,
   /{usr/,}bin/uname     rmix,
-  /usr/bin/xdg-open     Cxr -> sanitized_helper,
-  /usr/bin/xdg-email    Cxr -> sanitized_helper,
+  /usr/bin/xdg-open     rPUx,
+  /usr/bin/xdg-email    rPUx,
   /dev/null             rw,
   INSTDIR-program/uri-encode rmpux,
   /usr/share/libreoffice/share/config/* r,

From 1d3621dde6182f752ecf1e2d6f69a981960c2a16 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@rene-engelhard.de>
Date: Wed, 23 May 2018 21:12:30 +0200
Subject: [PATCH] deb#899380 apparmor: fix gpg encryption hang trying to lock
 random_seed

Change-Id: Ib9fb7652922dcc8364567953d17d7cae8ad170a7
---
 sysui/desktop/apparmor/program.soffice.bin | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index fb4c3f013d10..8ce7cbc171c1 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -190,5 +190,6 @@
    /usr/bin/gpgsm rm,
 
     owner @{HOME}/.gnupg/* r,
+    owner @{HOME}/.gnupg/random_seed rk,
   }
 }

From 6739a70d1af250814fe134c8752a058df433afc3 Mon Sep 17 00:00:00 2001
From: Rene Engelhard <rene@debian.org>
Date: Thu, 24 May 2018 11:47:55 +0200
Subject: [PATCH] apparmor: allow also /usr/lib{,32,64}/jvm/**/bin/java

since /usr/lib{,32,64}/jvm/**/jre/bin/java doesn't exist since OpenJDK 9

Change-Id: Ic9c052a756ba4a93595595da98148ff22169d333
Reviewed-on: https://gerrit.libreoffice.org/54754
Reviewed-by: Rene Engelhard <rene@debian.org>
Tested-by: Rene Engelhard <rene@debian.org>
---
 sysui/desktop/apparmor/program.soffice.bin | 1 +
 1 file changed, 1 insertion(+)

diff --git a/sysui/desktop/apparmor/program.soffice.bin b/sysui/desktop/apparmor/program.soffice.bin
index 24ff2fa854ac..87f6dff1d7c2 100644
--- a/sysui/desktop/apparmor/program.soffice.bin
+++ b/sysui/desktop/apparmor/program.soffice.bin
@@ -142,6 +142,7 @@ profile libreoffice-soffice INSTDIR-program/soffice.bin {
   /usr/lib{,32,64}/jvm/                         r,
   /usr/lib{,32,64}/jvm/**                       r,
   /usr/lib{,32,64}/jvm/**/jre/bin/java          mix,
+  /usr/lib{,32,64}/jvm/**/bin/java              mix,
   INSTDIR-**                        rw,
   INSTDIR-**.so                     m,
   INSTDIR-program/soffice.bin       mix,
-- 
2.17.0

